pkg/resource/pod_distruption_budget.go

```diff
@@ -0,0 +1,94 @@
+package resource
+
+import (
+	"errors"
+	"fmt"
+
+	policy "k8s.io/api/policy/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// PdbBuilder models the PodDistruptionBudget that the
+// operator maintains.
+type PdbBuilder struct {
+	*Cluster
+
+	Selector map[string]string
+}
+
+// Build creates a policy.PodDisruptionBudget for the
+// StatefulSet.
+func (b PdbBuilder) Build(obj runtime.Object) error {
+	pdb, ok := obj.(*policy.PodDisruptionBudget)
+	if !ok {
+		return errors.New("failed to cast to PDB object")
+	}
+
+	// if we only have one Node we cannot have a PDB
+	// TODO we need to validate this in the CRD API
+	if b.Spec().Nodes == 1 {
+		return nil
+	}
+
+	// TODO fix or should we use this?
+	if pdb.ObjectMeta.Name == "" {
+		pdb.ObjectMeta.Name = b.DiscoveryServiceName()
+	}
+
+	if pdb.ObjectMeta.Labels == nil {
+		pdb.ObjectMeta.Labels = map[string]string{}
+	}
+
+	// TODO test sevice name
+	// Can we use the common label selector?
```

Comment:
Seems desirable as running two crdb clusters in the same CRDB cluster couldn't currently have multiple PDBs if I'm reading this correctly?

Reply:
What do you mean?  There is a func that provides a common label, and I think I can reuse that.  I need to poke around a bit more to see how to convert a map into a label.

Reply:
```yaml
   app.kubernetes.io/instance: crdb
```

But to be honest we should have a

```yaml
  app.cockroachlabs.com/instance: crdb
```

Or something unique like that. Yes looking at it now the implementation needs to change, but we also might want better labels.  Thoughts?

Reply:
It looks like the PDB selector is just `app=cockroachdb`. Won't that conflict if I create to cluster with different PDB configurations?

Reply:
@chrisseto I got yah now.  That is bad code.  We need to decide which labels I am matching, which is the discussion above.

Reply:
All of this is updated now.  Still testing, but PTAL